lib/logger/lib/logger.ex

```diff
@@ -573,7 +573,7 @@ defmodule Logger do
     if compare_levels(level, min_level) != :lt do
       macro_log(level, data, metadata, caller)
     else
-      :ok
+      handle_unused_variable_warnings(data)
```

Comment:
Personally, I like `fn -> ... end` idea as it works nicely with what we do already today: simply ignore calls with lover level than `Logger.level`, and purging feels quite the same.